### PR TITLE
Add CNN1D limitation for stride > filter, add negative tests, simplify

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Add CNN1D limitation for stride > filter
 - add negative tests
 - simplify

### Tickets:
 - 55616

### CODE REVIEW

File: src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp [hunk 1]
```diff
@@ -214,6 +226,7 @@ GNA_NEG_INSTANTIATE(InputW, Fine, InvalidInputW, "Unsupported input width")
 GNA_NEG_INSTANTIATE(InputC, Fine, InvalidInputC, "Unsupported number of input channels")
 GNA_NEG_INSTANTIATE(Padding, InvalidPadding, Fine, "Convolution's input padding is not supported")
 GNA_NEG_INSTANTIATE(Stride, InvalidStride, Fine, "Unsupported convolution stride shape")
+GNA_NEG_INSTANTIATE(Stride1D, InvalidStride1D, Fine1D, "Unsupported effective convolution 1D stride shape")
```
Comment (kbruniec) [inline src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp:229]:
```suggestion
// GNA_NEG_INSTANTIATE(Stride1D, InvalidStride1D, Fine1D, "Unsupported effective convolution 1D stride shape")
```

File: src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution.cpp [hunk 2]
```diff
@@ -41,6 +41,7 @@ const std::vector<std::vector<size_t >> kernelsH1 = {{1, 3},
                                                      {1, 5}};
 const std::vector<std::vector<size_t >> stridesH1 = {{1, 1},
                                                      {1, 3}};
+const std::vector<std::vector<size_t>> stridesH1Big = {{1, 16}};
```
Comment (sirzabek) [inline src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution.cpp:44]: Maybe testing stride != multiple of 8 would be beneficial?

Comment (kbruniec) [reply src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution.cpp]: Added

File: src/plugins/intel_gna/gna_graph_compiler.cpp [hunk 3]
```diff
@@ -264,17 +264,18 @@ void GNAGraphCompiler::ConvolutionPrimitive(InferenceEngine::CNNLayerPtr layer)
         std::swap(convolution._padding_x, convolution._padding_y);
         std::swap(convolution._stride_x, convolution._stride_y);
         std::swap(convolution._dilation_x, convolution._dilation_y);
-    }
-
-    // Map 2d convolution to 1d if it's possible
-    if (GNAConvolutionLayer::isMappableFrom2DTo1D(in_height, in_width, convolution._kernel_x, convolution._stride_x)) {
+    } else if (GNAConvolutionLayer::isMappableFrom2DTo1D(in_height, in_width, convolution._kernel_x, convolution._stride_x)) {
+        // Map 2d convolution to 1d if it's possible
+        convolution._stride_x = (convolution._stride_y * in_width);
```
Comment (nkogteva) [inline src/plugins/intel_gna/gna_graph_compiler.cpp:269]: This PR has conflicts with https://github.com/openvinotoolkit/openvino/pull/9474 to support new format of kaldi IRs. Let's start with my PR to unlock https://github.com/openvinotoolkit/openvino/pull/5744

Comment (kbruniec) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:269]: Conflicting part is not significant, I could revert some part from my PR, or I can help you to resolve these conflicts from yours.

Comment (kbruniec) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: @nkogteva  conflicting part was removed

Comment (kbruniec) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: @nkogteva could you check if it is OK for you?

Comment (nkogteva) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: @kbruniec I think we need to run AC validation for this PR

Comment (nkogteva) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: @kbruniec I think we need to run AC validation for this PR

Comment (kbruniec) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: @nkogteva It may be worth to do so. But please note this snippet is outdated.

**Review by sirzabek — APPROVED** [review]

File: src/plugins/intel_gna/gna_graph_compiler.cpp (hunk 3 above)
Comment (nkogteva) [reply src/plugins/intel_gna/gna_graph_compiler.cpp]: Accuracy results looks ok

**Review by nkogteva — APPROVED** [review]

**Pull request merged**